examples/3d_examples/OctopusSwimming/set_environment_tapered_arm.py

```diff
@@ -0,0 +1,365 @@
+from collections import defaultdict
+import numpy as np
+import elastica as ea
+from elastica._calculus import _isnan_check
+
+from coomm.actuations.muscles import (
+    force_length_weight_poly,
+)
+from coomm.actuations.muscles import (
+    MuscleGroup,
+    LongitudinalMuscle,
+    ObliqueMuscle,
+    TransverseMuscle,
+    ApplyMuscleGroups,
+)
+
+
+class BaseSimulator(
+    ea.BaseSystemCollection,
+    ea.Constraints,
+    ea.Connections,
+    ea.Forcing,
+    ea.CallBacks,
+    ea.Damping,
+):
+    pass
+
+
+class OctopusArmCallBack(ea.CallBackBaseClass):
+    """
+    Call back function for octopus arm
+    """
+
+    def __init__(self, step_skip: int, callback_params: dict):
+        ea.CallBackBaseClass.__init__(self)
+        self.every = step_skip
+        self.callback_params = callback_params
+
+    def make_callback(self, system, time, current_step: int):
+        if current_step % self.every == 0:
+            self.callback_params["time"].append(time)
+            self.callback_params["step"].append(current_step)
+            self.callback_params["position"].append(system.position_collection.copy())
+            self.callback_params["velocity"].append(system.velocity_collection.copy())
+            self.callback_params["radius"].append(system.radius.copy())
+
+            return
+
+
+class ArmEnvironment:
```

Comment:
octopus environment? reason for calling arm environment?

Reply:
4ce37ff